python/ray/autoscaler/v2/utils.py

```diff
@@ -539,6 +542,29 @@ def _pending_node_report(data: ClusterStatus) -> str:
             return "\n".join(pending_lines)
         return " (no pending nodes)"
 
+    @staticmethod
+    def _constraint_report(data: ClusterStatus) -> str:
```

Comment:
only pass `data.resource_demands.cluster_constraint_demand` into this function.

Reply:
DONE